pages/docs/tutorials/native/apple-framework.md

```diff
@@ -0,0 +1,284 @@
+---
+type: tutorial
+layout: tutorial
+title:  "Kotlin/Native as an Apple Framework"
+description: "Compiling Kotlin/Native code and use it from Objective-C and Swift"
+authors: Eugene Petrenko
+date: 2018-08-05
+showAuthorInfo: false
+issue: EVAN-5132
+---
+
+Kotlin/Native provides bidirectional interoperability with Objective-C/Swift. 
+Objective-C frameworks and libraries can be used in Kotlin code.
+Kotlin module can be used in Swift/Objective-C code too.
+
+Besides that, Kotlin/Native has C interop. You may want to
+see the [Kotlin/Native as a Dynamic Library](dynamic-libraries.html)
+tutorial too.
+
+In that tutorial, we will see how to use Kotlin/Native code from
+Objective-C and Swift applications on macOS and iOS.
+We will build a Framework from a Kotlin code.
+
+In the tutorial we'll: 
+- [Create a Kotlin Library](#creating-a-kotlin-library) and compile it to Apple Framework
+- Examine generated [Objective-C and Swift API](#generated-framework-headers)
+- Use the Framework from [Objective-C](#using-the-code-from-objective-c) and [Swift](#using-the-code-from-swift)
+
+   
+## Creating a Kotlin Library
+
+Kotlin/Native compiler is able to produce a Framework for macOS and iOS
+out of the Kotlin code. The produced framework contains all declarations
+and binaries to use it with Objective-C and Swift
+
+The best way to understand the techniques is to try those techniques. 
+Let's create a tiny Kotlin library first and use it from an Objective-C program.
+
+We create a `lib.kt` file with the library contents:
+
+<div class="sample" markdown="1" mode="kotlin" theme="idea" data-highlight-only="1" auto-indent="false">
+
+```kotlin
+package demo
+
+object DemoObject {
+  val field = "A"
+}
+
+class DemoClass {
+  fun foo() : Long = 42
+}
+
+fun ints(b: Byte, s: Short, i: Int, l:Long) { }
+fun floats(f: Float, d: Double) { }
+
+fun strings(str: String) : String {
+  return "That is '$str' from C"
+}
+
+val globalString = "A global String"
+```
+</div>
+
+We need to have a Kotlin/Native compiler on our machines. 
+You may have a look at the
+[A Basic Kotlin/Native Application](basic-kotlin-native-app.html#obtaining-the-compiler)
+tutorial for more information on performing this step.
+Let's assume we have a console, where `kotlinc` command is available. 
+
+Now we call the following commands to compile the code into Frameworks
+for macOS, iOS and iOS emulator respectively:
+```bash
+kotlinc lib.kt -produce dynamic -output macOS/Demo
```

Comment:
Why not produce `framework` for macOS too?

Reply:
it is on the list of 3 commands, what do I miss?

Reply:
Your first command produces dynamic library (`macOS/Demo.dylib`), while other commands produce frameworks (`iOS*/Demo.framework`). Is it OK?

Reply:
fixed. It was a typoe